File: backend/test/java/de/griefed/serverpackcreator/utilities/common/JarUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.common;

import java.io.File;
import java.util.HashMap;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class JarUtilitiesTest {

  JarUtilities jarUtilities;

  JarUtilitiesTest() {
    jarUtilities = new JarUtilities();
  }

  @Test
  void copyFileFromJarTest() {
    jarUtilities.copyFileFromJar("banner.txt", JarUtilitiesTest.class,
                                 new File("tests").getAbsolutePath());
    Assertions.assertTrue(new File("tests/banner.txt").isFile());
  }

  @Test
  void getApplicationHomeForClassTest() {
    Assertions.assertNotNull(jarUtilities.getApplicationHomeForClass(JarUtilitiesTest.class));
  }

  @Test
  void systemInformationTest() {
    HashMap<String, String> system =
        jarUtilities.jarInformation(
            jarUtilities.getApplicationHomeForClass(JarUtilitiesTest.class));
    Assertions.assertNotNull(system);
    Assertions.assertNotNull(system.get("jarPath"));
    Assertions.assertTrue(system.get("jarPath").length() > 0);
    Assertions.assertNotNull(system.get("jarName"));
    Assertions.assertTrue(system.get("jarName").length() > 0);
    Assertions.assertNotNull(system.get("javaVersion"));
    Assertions.assertTrue(system.get("javaVersion").length() > 0);
    Assertions.assertNotNull(system.get("osArch"));
    Assertions.assertTrue(system.get("osArch").length() > 0);
    Assertions.assertNotNull(system.get("osName"));
    Assertions.assertTrue(system.get("osName").length() > 0);
    Assertions.assertNotNull(system.get("osVersion"));
    Assertions.assertTrue(system.get("osVersion").length() > 0);
  }

  @Test
  void copyFolderFromJarTest() {
    try {
      jarUtilities.copyFolderFromJar(
          JarUtilitiesTest.class,
          "/de/griefed/resources/manifests",
          "tests/manifestTest",
          "",
          "xml|json");
    } catch (Exception ignored) {
    }
    Assertions.assertTrue(new File("tests/manifestTest").isDirectory());
    Assertions.assertTrue(new File("tests/manifestTest/fabric-installer-manifest.xml").isFile());
    Assertions.assertTrue(new File("tests/manifestTest/mcserver").isDirectory());
    Assertions.assertTrue(new File("tests/manifestTest/mcserver/1.8.2.json").isFile());
  }
}

File: serverpackcreator-api/src/test/kotlin/de/griefed/serverpackcreator/api/common/JarUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.utilities.common.JarUtilities
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File

class JarUtilitiesTest internal constructor() {

    @Test
    fun copyFileFromJarTest() {
        JarUtilities.copyFileFromJar(
            "banner.txt", JarUtilitiesTest::class.java,
            File("tests").absolutePath
        )
        Assertions.assertTrue(File("tests/banner.txt").isFile)
    }

    @Test
    fun systemInformationTest() {
        val system: HashMap<String, String> = JarUtilities.jarInformation(JarUtilitiesTest::class.java)
        Assertions.assertNotNull(system)
        Assertions.assertNotNull(system["jarPath"])
        Assertions.assertTrue(system["jarPath"]!!.isNotEmpty())
        Assertions.assertNotNull(system["jarName"])
        Assertions.assertTrue(system["jarName"]!!.isNotEmpty())
        Assertions.assertNotNull(system["javaVersion"])
        Assertions.assertTrue(system["javaVersion"]!!.isNotEmpty())
        Assertions.assertNotNull(system["osArch"])
        Assertions.assertTrue(system["osArch"]!!.isNotEmpty())
        Assertions.assertNotNull(system["osName"])
        Assertions.assertTrue(system["osName"]!!.isNotEmpty())
        Assertions.assertNotNull(system["osVersion"])
        Assertions.assertTrue(system["osVersion"]!!.isNotEmpty())
    }

    @Test
    fun copyFolderFromJarTest() {
        try {
            JarUtilities.copyFolderFromJar(
                JarUtilitiesTest::class.java,
                "/de/griefed/resources/manifests",
                "tests/manifestTest",
                "",
                ".*\\.(xml|json)".toRegex()
            )
        } catch (ignored: Exception) {
        }
        Assertions.assertTrue(File("tests/manifestTest").isDirectory)
        Assertions.assertTrue(File("tests/manifestTest/fabric-installer-manifest.xml").isFile)
        Assertions.assertTrue(File("tests/manifestTest/mcserver").isDirectory)
        Assertions.assertTrue(File("tests/manifestTest/mcserver/1.8.2.json").isFile)
    }
}